### ISSUE TITLE
Match fails because `Could not find option for key :platform`

### ISSUE THREAD

**Issue by arikis** [fastlane/fastlane#9702]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

`match` fails because of the following error regardless of whether I set the `platform` value:

```
Could not find option for key :platform. Available keys: git_url, git_branch, type, app_identifier, username, keychain_name, keychain_password, readonly, team_id, team_name, verbose, force, skip_confirmation, shallow_clone, workspace, force_for_new_devices, skip_docs
```

It worked perfectly in `2.37.0`. I can see that docs say that there is a `platform` option in `match`, however, fastlane will strip it because:

```
Could not find option 'platform' in the list of available options
```

### Captured Output

Command Used: `ios nightly dev:true`
<details><summary>Output/Log</summary>

```

Successfully loaded Appfile at path '/Users/user/Developer/app/fastlane/Appfile'
- apple_id: 'redacted@r.com'
- team_name: 'Redacted'
-------
INFO [2017-07-07 13:40:13.13]: Successfully loaded custom action '/Users/some-user/Developer/Redacted/fastlane/actions/generate_dingbatz.rb'.
INFO [2017-07-07 13:40:13.14]: Successfully loaded custom action '/Users/some-user/Developer/Redacted/fastlane/actions/get_localized_strings.rb'.
INFO [2017-07-07 13:40:13.14]: Successfully loaded custom action '/Users/some-user/Developer/Redacted/fastlane/actions/increment_version_number_at_index.rb'.
INFO [2017-07-07 13:40:13.14]: Successfully loaded custom action '/Users/some-user/Developer/Redacted/fastlane/actions/mission_control.rb'.
INFO [2017-07-07 13:40:13.14]: Successfully loaded custom action '/Users/some-user/Developer/Redacted/fastlane/actions/write_string_to_file.rb'.
DEBUG [2017-07-07 13:40:13.15]: Checking if there are any plugins that should be loaded...
DEBUG [2017-07-07 13:40:13.23]: Using dev: true
INFO [2017-07-07 13:40:13.25]: -------------------------------------------------
INFO [2017-07-07 13:40:13.25]: --- Step: Verifying required fastlane version ---
INFO [2017-07-07 13:40:13.25]: -------------------------------------------------
INFO [2017-07-07 13:40:13.25]: Your fastlane version 2.45.0 matches the minimum requirement of 2.0.3  ✅
INFO [2017-07-07 13:40:13.25]: ------------------------------
INFO [2017-07-07 13:40:13.25]: --- Step: default_platform ---
INFO [2017-07-07 13:40:13.25]: ------------------------------
INFO [2017-07-07 13:40:13.26]: Driving the lane 'ios nightly' 🚀
INFO [2017-07-07 13:40:13.27]: --------------------------------------------------
INFO [2017-07-07 13:40:13.27]: --- Step: Switch to ios cert_inhouse_beta lane ---
INFO [2017-07-07 13:40:13.27]: --------------------------------------------------
INFO [2017-07-07 13:40:13.27]: Cruising over to lane 'ios cert_inhouse_beta' 🚖
INFO [2017-07-07 13:40:13.27]: -------------------------------------------------
INFO [2017-07-07 13:40:13.27]: --- Step: Switch to ios match_enterprise lane ---
INFO [2017-07-07 13:40:13.27]: -------------------------------------------------
INFO [2017-07-07 13:40:13.27]: Cruising over to lane 'ios match_enterprise' 🚖
INFO [2017-07-07 13:40:13.27]: -------------------
INFO [2017-07-07 13:40:13.27]: --- Step: match ---
INFO [2017-07-07 13:40:13.27]: -------------------
Successfully loaded Appfile at path '/Users/some-user/Developer/Redacted/fastlane/Appfile'
- apple_id: 'redacted@r.com'
- team_name: 'Redacted'
- app_identifier: 'com.redacted.app.developer'
- team_id: 'xxxxxxxxxx'
-------
ERROR [2017-07-07 13:40:13.28]: Could not find option 'platform' in the list of available options: git_url, git_branch, type, app_identifier, username, keychain_name, keychain_password, readonly, team_id, team_name, verbose, force, skip_confirmation, shallow_clone, workspace, force_for_new_devices, skip_docs, thus removing it
DEBUG [2017-07-07 13:40:13.28]: Taking value for 'team_name' from environment variable 'FASTLANE_TEAM_NAME'

+-----------------------+--------------------------------------+
|                   Summary for match 1.0.0                    |
+-----------------------+--------------------------------------+
| git_url               | git@github.com:redacted/ios-enterpri  |
|                       | se-certificates.git                  |
| team_id               | xxxxxxxxxx                           |
| app_identifier        | com.someapp.app.developer            |
| type                  | enterprise                           |
| readonly              | true                                 |
| force_for_new_devices | false                                |
| git_branch            | master                               |
| username              | redacted@r.com                       |
| keychain_name         | login.keychain                       |
| team_name             | Redacted                             |
| verbose               | false                                |
| force                 | false                                |
| skip_confirmation     | false                                |
| shallow_clone         | false                                |
| skip_docs             | false                                |
+-----------------------+--------------------------------------+

ERROR [2017-07-07 13:40:13.29]: Enterprise profiles are currently not officially supported in _match_, you might run into issues
INFO [2017-07-07 13:40:13.29]: Cloning remote git repo...
INFO [2017-07-07 13:40:13.30]: ▸ Cloning into '/var/folders/9w/hd1hqlbj22b55j_75j62c9br0000gn/T/d20170707-50323-sum7wo'...
INFO [2017-07-07 13:40:14.67]: ▸ remote: Counting objects: 1465, done.[K
INFO [2017-07-07 13:40:17.21]: ▸ remote: Total 1465 (delta 0), reused 0 (delta 0), pack-reused 1465[K
INFO [2017-07-07 13:40:17.21]: ▸ Receiving objects: 100% (1465/1465), 6.72 MiB | 2.48 MiB/s, done.
INFO [2017-07-07 13:40:17.22]: ▸ Resolving deltas: 100% (209/209), done.
INFO [2017-07-07 13:40:17.90]: 🔓  Successfully decrypted certificates repo
INFO [2017-07-07 13:40:17.90]: Installing certificate...
INFO [2017-07-07 13:40:17.90]: $ security list-keychains -d user
INFO [2017-07-07 13:40:17.91]: ▸     "/Users/some-user/Library/Keychains/login.keychain-db"
INFO [2017-07-07 13:40:17.91]: $ security find-certificate -c 'Apple Worldwide Developer Relations Certification Authority' /Users/some-user/Library/Keychains/login.keychain-db
INFO [2017-07-07 13:40:17.94]: ▸ keychain: "/Users/some-user/Library/Keychains/login.keychain-db"
INFO [2017-07-07 13:40:17.94]: ▸ version: 512
INFO [2017-07-07 13:40:17.94]: ▸ class: 0x80001000 
INFO [2017-07-07 13:40:17.94]: ▸ attributes:
INFO [2017-07-07 13:40:17.94]: ▸     "alis"<blob>="Apple Worldwide Developer Relations Certification Authority"
INFO [2017-07-07 13:40:17.94]: ▸     "cenc"<uint32>=0x00000003 
INFO [2017-07-07 13:40:17.94]: ▸     "ctyp"<uint32>=0x00000001 
INFO [2017-07-07 13:40:17.94]: ▸     "hpky"<blob>=0x88271709A9B618608BECEBBAF64759C55254A3B7  "\210'\027\011\251\266\030`\213\354\353\272\366GY\305RT\243\267"
INFO [2017-07-07 13:40:17.94]: ▸     "issu"<blob>=0x3062310B300906035504061302555331133011060355040A130A4150504C4520494E432E31263024060355040B131D4150504C452043455254494649434154494F4E20415554484F52495459311630140603550403130D4150504C4520524F4F54204341  "0b1\0130\011\006\003U\004\006\023\002US1\0230\021\006\003U\004\012\023\012APPLE INC.1&0$\006\003U\004\013\023\035APPLE CERTIFICATION AUTHORITY1\0260\024\006\003U\004\003\023\015APPLE ROOT CA"
INFO [2017-07-07 13:40:17.94]: ▸     "labl"<blob>="Apple Worldwide Developer Relations Certification Authority"
INFO [2017-07-07 13:40:17.94]: ▸     "skid"<blob>=0x88271709A9B618608BECEBBAF64759C55254A3B7  "\210'\027\011\251\266\030`\213\354\353\272\366GY\305RT\243\267"
INFO [2017-07-07 13:40:17.94]: ▸     "snbr"<blob>=0x01DEBCC4396DA010  "\001\336\274\3049m\240\020"
INFO [2017-07-07 13:40:17.94]: ▸     "subj"<blob>=0x308196310B300906035504061302555331133011060355040A0C0A4170706C6520496E632E312C302A060355040B0C234170706C6520576F726C647769646520446576656C6F7065722052656C6174696F6E733144304206035504030C3B4170706C6520576F726C647769646520446576656C6F7065722052656C6174696F6E732043657274696669636174696F6E20417574686F72697479  "0\201\2261\0130\011\006\003U\004\006\023\002US1\0230\021\006\003U\004\012\014\012Apple Inc.1,0*\006\003U\004\013\014#Apple Worldwide Developer Relations1D0B\006\003U\004\003\014;Apple Worldwide Developer Relations Certification Authority"
DEBUG [2017-07-07 13:40:18.03]: Certificate 'xxxxxxxxxx.cer' is already installed on this machine

+-------------------+---------------------------------+
|                Installed Certificate                |
+-------------------+---------------------------------+
| User ID           | xxxxxxxxxx                      |
| Common Name       | iPhone Distribution: Redacted   |
| Organisation Unit | xxxxxxxxxx                      |
| Organisation      | Redacted                        |
| Country           | US                              |
| Start Datetime    | Oct 20 14:38:47 2016 GMT        |
| End Datetime      | Oct 20 14:38:47 2019 GMT        |
+-------------------+---------------------------------+

INFO [2017-07-07 13:40:18.20]: Installing provisioning profile...

+---------------------+--------------------------------------------------------+-----------------------------------------+
|                                             Installed Provisioning Profile                                             |
+---------------------+--------------------------------------------------------+-----------------------------------------+
| Parameter           | Environment Variable                                   | Value                                   |
+---------------------+--------------------------------------------------------+-----------------------------------------+
| App Identifier      |                                                        | com.someapp.app.developer               |
| Type                |                                                        | enterprise                              |
| Profile UUID        | sigh_com.someapp.app.developer_enterprise              | xxxxxxxx-xxxxxxxxxxxxxxxxxxxxxxxxxxx    |
| Profile Name        | sigh_com.someapp.app.developer_enterprise_profile-name | match InHouse com.someapp.app.developer |
| Development Team ID | sigh_com.someapp.app.developer_enterprise_team-id      | xxxxxxxxxx                              |
+---------------------+--------------------------------------------------------+-----------------------------------------+

INFO [2017-07-07 13:40:18.30]: All required keys, certificates and provisioning profiles are installed 🙌
INFO [2017-07-07 13:40:18.31]: 🔒  Successfully encrypted certificates repo
INFO [2017-07-07 13:40:18.31]: Setting Provisioning Profile type to 'enterprise'
ERROR [2017-07-07 13:40:18.34]: Unable to send the crash report.
ERROR [2017-07-07 13:40:18.34]: Please open an issue on GitHub if you need help!
ERROR [2017-07-07 13:40:18.34]: Lane 'match_enterprise' failed with exception: Could not find option for key :platform. Available keys: git_url, git_branch, type, app_identifier, username, keychain_name, keychain_password, readonly, team_id, team_name, verbose, force, skip_confirmation, shallow_clone, workspace, force_for_new_devices, skip_docs
WARN [2017-07-07 13:40:18.35]: Lane Context:
INFO [2017-07-07 13:40:18.35]: {:DEFAULT_PLATFORM=>:ios, :PLATFORM_NAME=>:ios, :LANE_NAME=>"ios nightly", :SIGH_PROFILE_TYPE=>"enterprise"}
ERROR [2017-07-07 13:40:18.35]: Could not find option for key :platform. Available keys: git_url, git_branch, type, app_identifier, username, keychain_name, keychain_password, readonly, team_id, team_name, verbose, force, skip_confirmation, shallow_clone, workspace, force_for_new_devices, skip_docs
INFO [2017-07-07 13:40:18.35]: Successfully generated documentation at path '/Users/some-user/Developer/Redacted/fastlane/README.md'

+------+-----------------------+-------------+
|              fastlane summary              |
+------+-----------------------+-------------+
| Step | Action                | Time (in s) |
+------+-----------------------+-------------+
| 1    | Verifying required    | 0           |
|      | fastlane version      |             |
| 2    | default_platform      | 0           |
| 3    | Switch to ios         | 0           |
|      | cert_inhouse_beta     |             |
|      | lane                  |             |
| 4    | Switch to ios         | 0           |
|      | match_enterprise lane |             |
| 💥   | match                 | 5           |
+------+-----------------------+-------------+

ERROR [2017-07-07 13:40:18.36]: fastlane finished with errors
ERROR [2017-07-07 13:40:18.39]: Unable to send the crash report.
ERROR [2017-07-07 13:40:18.39]: Please open an issue on GitHub if you need help!


```

</details>

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.12.5                                     |
| Ruby                        | 2.1.0                                       |
| Bundler?                    | true                                        |
| Git                         | git version 2.11.0 (Apple Git-81)           |
| Installation Source         | ~/.rvm/gems/ruby-2.1.0/bin/fastlane         |
| Host                        | Mac OS X 10.12.5 (16F73)                    |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.1.0/lib                |
| OpenSSL Version             | OpenSSL 1.0.1j 15 Oct 2014                  |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 8.3.3                                       |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
fastlane_version "2.0.3"
default_platform :ios

require_relative 'redacted_fastlane'

platform :ios do
  before_all do |lane, options|
    next if options[:dry_run] || options[:dev]
    my_ensure_git_status_clean
  end

  lane :cert_store do |options|
    match_non_enterprise(options)
  end

  lane :cert_development do |options|
    match_enterprise(options)
  end

  lane :cert_inhouse_beta do |options|
    match_enterprise(options.merge({
      type: $match_type_enterprise
    }))
  end

  lane :cert_samples do |options|
    match_enterprise(options.merge({
      app_identifier: $app_identifier_wildcard_samples
    }))
  end
  
  lane :certs do |options|
    begin
      cert_store(options)
      cert_testflight(options)
      cert_development(options)
      cert_inhouse_beta(options)
      cert_samples(options)
    rescue Interrupt
      cleanup(options)
    end
  end

  lane :nightly do |options|
    begin
      cert_inhouse_beta(options) # Make sure we are using latest certs and provisioning profile
      get_localized_strings
      configure_arbitrary_loads(options.merge({ allow_arbitrary_loads: true }))
      ipa_name = my_gym(options.merge({
        scheme: $scheme_inhouse_beta
      }))
      mission_control(options.merge({ ipa_name: ipa_name }))
    rescue Interrupt
      cleanup_on_interrupt
    end
  end

  after_all do |lane, options|
    # Only called if executed lane was successful
    UI.message "Lane '#{lane}' finished successfully"
  end

  error do |lane, exception, options|
    UI.error "Lane '#{lane}' failed with exception: #{exception}"
    cleanup(options)
  end

  def cleanup(options)
    return if options[:dev]
    clean_build_artifacts # Remove .ipa, .dSYM etc
    my_git_reset # Reset changes in an other files
    restore_original_git_branch
  end

  private_lane :match_enterprise do |options|
    my_match(options.merge({
      app_identifier: options[:app_identifier] || $app_identifier_inhouse_beta
    }))
  end

  private_lane :match_non_enterprise do |options|
    my_match(options.merge({
      type: options[:type] || $match_type_appstore,
      team_id: options[:team_id] || $team_id_non_enterprise,
      app_identifier: options[:app_identifier] || $app_identifier_appstore,
      git_url: options[:git_url] || $git_url_appstore
    }))
  end
end```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
# Variables used across fastlane files
$team_id_enterprise = 'xxxxxxxxxx'
$team_id_non_enterprise = 'yyyyyyyyyy'
$app_identifier_testflight = 'com.someapp.beta'
$app_identifier_inhouse_beta = 'com.someapp.app.developer'
$app_identifier_appstore = 'com.someapp.app'

# Common for all lanes
apple_id 'redacted@r.com'
team_name 'Redacted  '

# Lane specific
for_lane :beta_inhouse do
  app_identifier $app_identifier_inhouse_beta
  team_id $team_id_enterprise
end

for_lane :nightly do
  app_identifier $app_identifier_inhouse_beta
  team_id $team_id_enterprise
end

for_lane :store do
  app_identifier $app_identifier_appstore
  team_id $team_id_non_enterprise
end

for_lane :beta_testflight do
  app_identifier $app_identifier_testflight
  team_id $team_id_non_enterprise
end
```

```ruby
#my_match file

$match_type_development = 'development'
$match_type_appstore = 'appstore'
$match_type_enterprise = 'enterprise'
$git_url_enterprise = 'git@github.com:redacted/enterprise-certificates.git'
$git_url_appstore = 'git@github.com:redacted/appstore-certificates.git'

def my_match(options)
  team_id = options[:team_id] || $team_id_enterprise
  app_id = options[:app_identifier] || $app_identifier_inhouse_beta
  git_url = options[:git_url] || $git_url_enterprise
  type = options[:type] || $match_type_development
  force_for_new_devices = options.b(:force_for_new_devices)
  readonly = options.b(:readonly, d: true)

  if force_for_new_devices
    readonly = false
  end

  if type.eql? $match_type_enterprise
    ENV[env_key_enterprise] = "true"
  end

  match(
    git_url: git_url,
    team_id: team_id,
    app_identifier: app_id,
    type: type,
    readonly: readonly,
    force_for_new_devices: force_for_new_devices,
    platform: 'ios'
  )
  ENV[env_key_enterprise] = nil
end

def env_key_enterprise
  unless @env_key_enterprise
    @env_key_enterprise = 'MATCH_FORCE_ENTERPRISE'
  end
  @env_key_enterprise
end

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.45.0  | ✅ Up-To-Date  |
| deliver  | 2.0.1   | ✅ Up-To-Date  |
| gym      | 2.0.0   | ✅ Up-To-Date  |
| match    | 1.0.0   | ✅ Up-To-Date  |
| scan     | 1.0.0   | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| rake                      | 12.0.0       |
| json                      | 1.8.6        |
| mini_portile2             | 2.1.0        |
| nokogiri                  | 1.7.1        |
| aws-sdk-v1                | 1.67.0       |
| i18n                      | 0.8.1        |
| minitest                  | 5.10.1       |
| thread_safe               | 0.3.6        |
| tzinfo                    | 1.2.3        |
| activesupport             | 4.2.8        |
| claide                    | 1.0.2        |
| fuzzy_match               | 2.0.4        |
| nap                       | 1.1.0        |
| cocoapods-core            | 1.2.1        |
| cocoapods-deintegrate     | 1.0.1        |
| cocoapods-downloader      | 1.1.3        |
| cocoapods-plugins         | 1.0.0        |
| cocoapods-search          | 1.0.0        |
| cocoapods-stats           | 1.0.0        |
| netrc                     | 0.7.8        |
| cocoapods-trunk           | 1.2.0        |
| cocoapods-try             | 1.1.0        |
| colored2                  | 3.1.2        |
| escape                    | 0.0.4        |
| fourflusher               | 2.0.1        |
| gh_inspector              | 1.0.3        |
| molinillo                 | 0.5.7        |
| ruby-macho                | 1.1.0        |
| CFPropertyList            | 2.3.5        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.0        |
| cocoapods                 | 1.2.1        |
| curb                      | 0.9.3        |
| colored                   | 1.2          |
| fastimage                 | 2.1.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.1        |
| babosa                    | 1.0.2        |
| bundler                   | 1.15.1       |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.5        |
| dotenv                    | 2.2.1        |
| excon                     | 0.57.1       |
| multipart-post            | 2.0.0        |
| faraday                   | 0.12.1       |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.11.0.1     |
| jwt                       | 1.5.6        |
| little-plugger            | 1.1.4        |
| multi_json                | 1.12.1       |
| logging                   | 2.2.2        |
| memoist                   | 0.16.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| declarative               | 0.0.9        |
| declarative-option        | 0.1.0        |
| uber                      | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.0.2        |
| google-api-client         | 0.12.0       |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| plist                     | 3.3.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| slack-notifier            | 1.5.1        |
| terminal-notifier         | 1.8.0        |
| unicode-display_width     | 1.3.0        |
| terminal-table            | 1.8.0        |
| tty-screen                | 0.5.0        |
| word_wrap                 | 1.0.0        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.2.8        |
| xcpretty-travis-formatter | 0.0.4        |
| facets                    | 3.1.0        |
| require_all               | 1.4.0        |
| ast                       | 2.3.0        |
| parser                    | 2.4.0.0      |
| powerpack                 | 0.1.1        |
| rainbow                   | 2.2.2        |
| ruby-progressbar          | 1.8.1        |
| rubocop                   | 0.48.1       |
</details>

*generated on:* **2017-07-07**
</details>

**Comment by KrauseFx** [comment]
Hey @arikis, make sure to remove `deliver`, `gym`, `match` and `scan` from your Gemfile and your system `[sudo] gem uninstall match`, as you seem to have some pretty old versions installed :) 

Let me know if that works for you

**Comment by arikis** [comment]
@KrauseFx thanks for the fast response, that solved the issue :)

**Comment by Sajjon** [comment]
I too had this issue. I am not sure if this will help anyone, but I was using rbenv and bundler. 

I had 

```ruby
gem 'match'
``` 
inside my `Gemfile`. **I removed this entry.**

I removed my `Gemfile.lock`

I ran `gem uninstall match`:

```bash
› gem uninstall match
Remove executables:
	match

in addition to the gem? [Yn]  Y
Removing match
Successfully uninstalled match-1.0.0
```

but running `which match` resulted in:
`/Users/sajjon/.rbenv/shims/match`

So I ran: `rbenv rehash` which fixed it for me, verified by `which match` -> `match not found`.

Then it worked.
